### ISSUE TITLE
Looks like most of indicator calculation is wrong. 

### ISSUE THREAD

**Issue by dipankar08** [anandanand84/technicalindicators#249]
I have tried to use this indicator and I found significant different the actual value and the values provided by this indicator module.  Looks like RSI, PSAR, MACD proving the wrong results. ( while examine with talib, tradingview ). I see only SMA is working as expected. 

I think the libs is not yet ready for prod usecase.

**Comment by da7a90** [comment]
> I have tried to use this indicator and I found significant different the actual value and the values provided by this indicator module. Looks like RSI, PSAR, MACD proving the wrong results. ( while examine with talib, tradingview ). I see only SMA is working as expected.
> 
> I think the libs is not yet ready for prod usecase.

I believe the same, I was using a library that depends on this one and the results are off and only sma and ema are working

**Comment by moshtaghi** [comment]
> 

Since the calculation is based on historical data and serially, to have an accurate result, the number of input candles must be increased to 1000. :)

**Comment by rlmomin** [comment]
> > 
> 
> Since the calculation is based on historical data and serially, to have an accurate result, the number of input candles must be increased to 1000. :)

I agree with his statement.  I use this library and use over 1000 candles.  The calculations match exactly with my charts on the TD Ameritrade ThinkorSwim platform.

**Comment by rishikesh2003** [comment]
Yes, I also agree. I've used most of the indicators in this library and I still don't find any errors in the value. You should give more input data in order for the calculations to be accurate

**Comment by DroidBoyJr** [comment]
Confirmed. it's the sample set size. For ema interval 50, I took history of 200 points and it showed correct value. If I take only 51 or so, it shows incorrect EMA calculation.

**Comment by saru2020** [comment]
it's a good library, where most indicators seem to be working correctly.

it is true that giving all candles/data works for some indicators like RSI and MACD(for SMA only) but not for all, or at least I see an issue with MACD(EMA based) even after feeding with all the candle/data as input, looks like EMA isn't working properly, are there any inputs to fix this issue?

**Comment by liesauer** [comment]
i feed 500 continuous candles to calculate Stochastic Oscillator, sometimes it looks good(still a little deviation), but it looks ridiculous in next second. only the last candle is realtime updating, how does it make such huge difference?
```
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 11.11111111111111, d: 37.16962962962969 }
{ k: 11.11111111111111, d: 37.16962962962969 }
{ k: 11.11111111111111, d: 37.16962962962969 }
{ k: 11.11111111111111, d: 37.16962962962969 }
{ k: 11.11111111111111, d: 37.16962962962969 }
{ k: 11.11111111111111, d: 37.16962962962969 }
{ k: 11.122222222222275, d: 37.17333333333341 }
{ k: 11.122222222222275, d: 37.17333333333341 }
{ k: 11.122222222222275, d: 37.17333333333341 }
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 77.77777777777779, d: 59.391851851851904 }
{ k: 77.77777777777779, d: 59.391851851851904 }
```

**Comment by liesauer** [comment]
i don't think it's the data samples issue.

![image](https://github.com/anandanand84/technicalindicators/assets/8676741/8fd9e243-74e0-4a0b-befe-b1b7063e7f12)
